MovingBlocks/Terasology
# Fixed auto movement

### Contains

A couple of additions for #2557.
- Fixed the jerky movement because of CharacterImpulseEvent. This approach simply simulates the forwards key pressed down.
- The character no more stops after jumping or strafing.
- Added code to simulate key presses. This can be extended to other tasks that require pressing of a key through code as if it was pressed on the keyboard.

### How to test

1. Start a game
2. Press R to start moving.
3. Confirm that character starts moving and stops when one of forward move button, backward move button or auto move button is pressed.

### Outstanding before merging

Nothing as of now. Let me know if there are any issues with this implementation while testing. I couldn't find any.
